h2o-core/src/main/java/water/rapids/Merge.java

```diff
@@ -327,26 +329,145 @@ public void map(Chunk[] cs, NewChunk[] ncs) {
+    int chunkLeftMSB[] = new int[numChunks];  // using too much space repeating the same value here, but, limited
+    int chunkRightMSB[] = new int[numChunks]; // leave it alone so as not to re-write chunkStitcher.
+    int chunkBatch[] = new int[numChunks];
+    int k = 0;
+    for( SortCombine thisbm : bmList ) {
+      if (thisbm._numRowsInResult == 0) continue;
+      int thisChunkSizes[] = thisbm._chunkSizes;
+      for (int j=0; j<thisChunkSizes.length; j++) {
+        chunkSizes[k] = thisChunkSizes[j];
+        chunkLeftMSB [k] = thisbm._leftSB._msb;
+        chunkRightMSB[k] = -1;
+        chunkBatch[k] = j;
+        k++;
+      }
+    }
+    Log.debug("took: " + (System.nanoTime() - t0) / 1e9 +" seconds.");
+
+    // Now we can stitch together the final frame from the raw chunks that were
+    // put into the store
+    Log.info("Allocating and populated espc ...");
+    t0 = System.nanoTime();
+    long espc[] = new long[chunkSizes.length+1];
+    int i=0;
+    long sum=0;
+    for (long s : chunkSizes) {
+      espc[i++] = sum;
+      sum+=s;
+    }
+    espc[espc.length-1] = sum;
+    Log.debug("took: " + (System.nanoTime() - t0) / 1e9+" seconds");
+    assert(sum==ansN);
+
+    Log.info("Allocating dummy vecs/chunks of the final frame ...");
+    t0 = System.nanoTime();
+    int numLeftCols = leftFrame.numCols();
+    int numColsInResult = numLeftCols;
+    final byte[] types = new byte[numColsInResult];
+    final String[][] doms = new String[numColsInResult][];
+    final String[] names = new String[numColsInResult];
+    for (int j=0; j<numLeftCols; j++) {
+      types[j] = leftFrame.vec(j).get_type();
+      doms[j] = leftFrame.domains()[j];
+      names[j] = leftFrame.names()[j];
+    }
+
+    Key<Vec> key = Vec.newKey();
+    Vec[] vecs = new Vec(key, Vec.ESPC.rowLayout(key, espc)).makeCons(numColsInResult, 0, doms, types);
+    Log.debug("took: " + (System.nanoTime() - t0) / 1e9+" seconds");
+
+    Log.info("Finally stitch together by overwriting dummies ...");
+    t0 = System.nanoTime();
+    Frame fr = new Frame(names, vecs);
+    ChunkStitcher ff = new ChunkStitcher(chunkSizes, chunkLeftMSB, chunkRightMSB, chunkBatch);
+    ff.doAll(fr);
+    Log.debug("took: " + (System.nanoTime() - t0) / 1e9+" seconds.");
+    return fr;
+  }
+  public static Frame sortOnly(final Frame leftFrame, final int leftCols[], int[][] id_maps, int[] ascendingL) {
+
+    createIndex(true ,leftFrame,leftCols,id_maps, ascendingL);  // sort the columns.
+    Log.info("Making BinaryMerge RPC calls ... ");
+    ArrayList<SortCombine> bmList = gatherSameMSBRows(leftFrame); // For each MSB, gather sorted rows with same MSB into one spot 
+    
+    Log.info("Allocating and populating chunk info (e.g. size and batch number) ...");
+    return allocatePopulateChunk(bmList, leftFrame);
+
+  }
+
+
+    private static RadixOrder createIndex(boolean isLeft, Frame fr, int[] cols, int[][] id_maps, int[] ascending) {
```

Comment:
formating

Reply:
Fixed

Reply:
it is not, the indent of the method header is wrong